Source/WebCore/dom/Document.cpp

```diff
@@ -4065,8 +4065,9 @@ const URL& Document::urlForBindings() const
             return false;
 
         auto areSameSiteIgnoringPublicSuffix = [](StringView domain, StringView otherDomain) {
-            auto domainString = topPrivatelyControlledDomain(domain.toStringWithoutCopying());
-            auto otherDomainString = topPrivatelyControlledDomain(otherDomain.toStringWithoutCopying());
+            auto& publicSuffixStore = PublicSuffixStore::singleton();
```

Comment:
I would recommend against using a singleton and instead hang an instance off the Page. Singletons, while appealing in their simplicity, make testing and reuse more challenging.

Reply:
The code was indeed called from various threads. However, the old code was thread safe, the new one is not (see my other comment). Public Suffix was always a global static map with a Lock. So using a singleton doesn't make things worse in that aspect (it was always global). However, I'm curious what you feel that benefits are to using a singleton vs keeping the current global static map?

Reply:
The goal of this change is to introduce a public suffix cache but use it only in certain case (web process in cocoa platforms). Having a singleton makes the configuration and management easier (compared to adding a global flag (or a global cache) for controlling the caching behavior).